README.md

```diff
@@ -1,79 +1,165 @@
+
+1. Open a shell and install dependecies by running:
+    ```bash
+    npm install
+    ```
+
+### Setup Environment Variables
+
+This setup uses a shared cloud Supabase database for simplicity.
+1. Run the command:
+   ```bash
+   cp .env.local.example .env.local
+   ```
+2. Request the `VITE_SUPABASE_URL` and `VITE_SUPABASE_ANON_KEY` from the team or go to settings > api from the Supabase Dashboard tp get them yourself. Other variables in `.env.local` can be ignored for now.
+
+### Run the application
+1. Run the command in your terminal from the project directory:
+```bash
+   npm run dev
+   ```
+
+## Testing
+
+- To run all tests: `npm run test`.
+
+## Setup Application with Local Database (Optional)
+
+This section is only needed if you want to isolate your database and not use a shared environment.
+
+### Configure Environmental Variables
+Local setup requires a few one-time steps. In the following steps, you will start Supabase locally and put values from Supabase into your .env.local file.
+
+1. Like before, create a copy of .env.local.example:
+
+   ```bash
+   cp .env.local.example .env.local
+   ```
+
+1. Open the .env.local file in VSCode. You will need to put values in this file after the next step, so keep it open.
+
+1. Install supabase CLI from [here](https://supabase.com/docs/guides/cli/getting-started?queryGroups=platform&platform=linux)
+
+1. Initialize Supabase locally. You will have to do this on the local system, not in the devcontainer since supabase will run in docker on the host. 
+
+    Open a *[local terminal window](https://stackoverflow.com/questions/59815283/open-local-terminal-in-vscode-when-running-in-ssh-mode?rq=1)* from the plymouth-housing folder and run: 
+   ```bash 
+   supabase init
+   ```
+
+1. Start Supabase locally. In the same local terminal window run: 
+
+   ```bash 
+   supabase start
+   ```
+
+    Enter "y" when/if asked to install the Supabase package.
+
+    Give it a moment to spin up the containers. When it's finished, it will display URLs and keys that need to go into your local environment file.
+    
+    Put the API URL after `VITE_SUPABASE_URL=` variable and the anon key after the `VITE_SUPABASE_ANON_KEY=` variable in the .env.local file. Double-check you copied the value correctly then save the file.
+
+    (These steps are derived from the [Original Supabase instructions here](https://supabase.com/docs/guides/cli/local-development#start-supabase-services))
+
+    _Note: If you have existing Supabase containers running, you may need to run `supabase stop --no-backup` to reset your environment. Careful, this will delete anything saved in your local database._
+
+1. Run `supabase stop` to close Supabase when you are done.
+
+#### Configure OAuth Environment Variables
+
+If you need to use OAuth, be sure to follow the steps listed in _.env.local.example_ to set the OAuth variables. Once the variables are set, run:
```

Comment:
Note: If you are trying to log in using Azure OAuth but get this error
{"code":400,"error_code":"validation_failed","msg":"Unsupported provider: provider is not enabled"}, you can add the following code to supabase/config.toml to enable the Azure OAuth provider in your local supabase.

```
# Use Azure OAuth provider
[auth.external.azure]
enabled = true
client_id = "env(SUPABASE_AUTH_AZURE_CLIENT_ID)"
# DO NOT commit your OAuth provider secret to git. Use environment variable substitution instead.
secret = "env(SUPABASE_AUTH_AZURE_SECRET)"
redirect_uri = "env(SUPABASE_AUTH_AZURE_REDIRECT_URI)"
url = `"env(SUPABASE_AUTH_AZURE_URI)"
```

Reply:
added for now, but will likely have to remove when migrating to Azure.